cmake/everest-generate.cmake

```diff
@@ -127,171 +133,229 @@ function (ev_add_modules)
+                    "$<TARGET_PROPERTY:generate_cpp_files,EVEREST_GENERATED_INCLUDE_DIR>"
+                    ${MODULE_LOADER_DIR}
+            )
+
+            target_sources(${MODULE_NAME}
+                PRIVATE
+                    ${MODULE_PATH}/${MODULE_NAME}.cpp
+                    "${MODULE_LOADER_DIR}/ld-ev.cpp"
+            )
+
+            target_link_libraries(${MODULE_NAME}
+                PRIVATE
+                    everest::framework
+            )
+
+            add_dependencies(${MODULE_NAME} ld-ev_${MODULE_NAME} generate_cpp_files)
+
+            install(TARGETS ${MODULE_NAME}
+                DESTINATION "${EVEREST_MODULE_INSTALL_PREFIX}/${MODULE_NAME}"
+            )
+
+            install(FILES ${MODULE_PATH}/manifest.yaml
+                DESTINATION "${EVEREST_MODULE_INSTALL_PREFIX}/${MODULE_NAME}"
+            )
+
+            list(APPEND MODULES ${MODULE_NAME})
+            add_subdirectory(${MODULE_PATH})
+        endif()
+    else()
+        message(WARNING "C++ module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
         return()
     endif()
 
-    message(STATUS "Setting up C++ module ${MODULE_NAME}")
-
-    get_target_property(GENERATED_OUTPUT_DIR generate_cpp_files EVEREST_GENERATED_OUTPUT_DIR)
-
-    set(GENERATED_MODULE_DIR "${GENERATED_OUTPUT_DIR}/modules")
-    set(MODULE_LOADER_DIR ${GENERATED_MODULE_DIR}/${MODULE_NAME})
-
-    add_custom_command(
-        OUTPUT
-            ${MODULE_LOADER_DIR}/ld-ev.hpp
-            ${MODULE_LOADER_DIR}/ld-ev.cpp
-        COMMAND
-            ${EV_CLI} module generate-loader
-                --disable-clang-format
-                --schemas-dir "$<TARGET_PROPERTY:generate_cpp_files,EVEREST_SCHEMA_DIR>"
-                --output-dir ${GENERATED_MODULE_DIR}
-                ${MODULE_NAME}
-        DEPENDS
-            manifest.yaml
-        WORKING_DIRECTORY
-            ${PROJECT_SOURCE_DIR}
-        COMMENT
-            "Generating ld-ev for module ${MODULE_NAME}"
+    # FIXME (aw): this will override EVEREST_MODULES, might not what we want
+    set_property(
+        GLOBAL
+        PROPERTY EVEREST_MODULES ${EVEREST_MODULES}
     )
+endfunction()
 
-    add_custom_target(ld-ev_${MODULE_NAME}
-        DEPENDS ${MODULE_LOADER_DIR}/ld-ev.cpp
-    )
+function (ev_add_js_module MODULE_NAME)
+    set(EVEREST_MODULE_INSTALL_PREFIX "${CMAKE_INSTALL_LIBEXECDIR}/everest/modules")
 
-    add_executable(${MODULE_NAME})
+    set(MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${MODULE_NAME}")
+
+    if(IS_DIRECTORY ${MODULE_PATH})
+        if(NOT ${EVEREST_ENABLE_JS_SUPPORT})
+            message(STATUS "Excluding JavaScript module ${MODULE_NAME} because EVEREST_ENABLE_JS_SUPPORT=${EVEREST_ENABLE_JS_SUPPORT}")
+            return()
+        elseif("${MODULE_NAME}" IN_LIST EVEREST_EXCLUDE_MODULES)
+            message(STATUS "Excluding module ${MODULE}")
+            return()
+        else()
+            message(STATUS "Setting up JavaScript module ${MODULE_NAME}")
```

Comment:
Don't know if you would agree with this style but in order to keep indentation low, one could invert the ifs like
```
if(NOT IS_DIRECTORY ..)
  warning ...
  return()
endif()

if (NOT ${EVEREST_ENABLE_JS_SUPPORT})
  warning ...
  return()
endif()

if ("${MODULE_NAME} IN_LIST ...)
  warning ...
  return()
endif()

# now the normal case ...
```
same would go the other add_ functions I guess